fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoansApiResourceSwagger.java

```diff
@@ -517,7 +517,11 @@ private GetLoansLoanIdFeeFrequency() {}
 
             public GetLoansLoanIdCurrency currency;
             @Schema(example = "1000000.000000")
+            public Double totalPrincipal;
+            @Schema(example = "1000000.000000")
```

Comment:
We should not work with Double, use BigDecimal. Since you touch it, fix it too Double -> BigDecimal ;)